lstchain/scripts/lstchain_mc_dl2_to_irf.py

```diff
@@ -0,0 +1,214 @@
+                    type=lambda x: bool(strtobool(x)), dest='irf_type',
+                    help='True for point-like IRF, False for Full Enclosure',
+                    default=True, required=False
+                    )
+
+parser.add_argument('--config', '-conf', type=Path,
+                    dest='config',
+                    help='Config file for selection cuts',
+                    default=None, required=False
+                    )
+
+args = parser.parse_args()
+
+def main():
+
+    output_dir = args.output_irf_dir.absolute()
+    output_dir.mkdir(exist_ok=True)
+
+    log.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    logging.getLogger().addHandler(handler)
+
+    if args.config is None:
+        cuts = read_configuration_file(os.path.join(os.path.dirname(__file__), '../data/data_selection_cuts.json'))
+    else:
+        cuts = read_configuration_file(args.config)
+
+    if args.gamma_diff_file is None:
+        particles = {
+	       "gamma": {
+	          "file": args.gamma_file,
+	          "target_spectrum": CRAB_HEGRA,
+	          },
+        #	"proton": {
+        #             "file": args.proton_file,
+        #	      "target_spectrum": IRFDOC_PROTON_SPECTRUM,
+        #             },
+        #	"electron": {
+        #             "file": args.electron_file,
+        #	      "target_spectrum": IRFDOC_ELECTRON_SPECTRUM,
+        #             },
+    	}
+    else:
+        #Using proton spectrum at the moment for unit tests
+        particles = {
+	       "gamma": {
+	          "file": args.gamma_diff_file,
+	          "target_spectrum": IRFDOC_PROTON_SPECTRUM,
+	          },
+        #	"proton": {
+        #             "file": args.proton_file,
+        #	      "target_spectrum": IRFDOC_PROTON_SPECTRUM,
+        #             },
+        #	"electron": {
+        #             "file": args.electron_file,
+        #	      "target_spectrum": IRFDOC_ELECTRON_SPECTRUM,
+        #             },
+    	}
+
+    for k, p in particles.items():
+        log.info(f"Simulated {k.title()} Events:")
+        p["events"], p["simulation_info"] = read_mc_dl2_to_pyirf(p["file"])
+        p["simulated_spectrum"] = PowerLaw.from_simulation(p["simulation_info"], 50 * u.hour)
+        p["events"]["weight"] = calculate_event_weights(
+            p["events"]["true_energy"], p["target_spectrum"], p["simulated_spectrum"]
+        )
+        p["events"]["source_fov_offset"] = calculate_source_fov_offset(p["events"])
+        # calculate theta / distance between reco and assumed source position
+        p["events"]["theta"] = calculate_theta(
+            p["events"],
+            assumed_source_az=p["events"]["true_az"],
+            assumed_source_alt=p["events"]["true_alt"],
+        )
+        log.info(p["simulation_info"])
+
+    gammas = particles["gamma"]["events"]
+    # selecting 1 tel_id as the data is collected from only 1 for now
+    gammas = gammas[gammas["tel_id"]==1]
```

Comment:
What if people want to use this script for MAGIC / LST observations. Shouldn't this at least be configurable?